core/src/main/scala/kafka/log/TimeIndex.scala

```diff
@@ -55,6 +55,7 @@ class TimeIndex(_file: File, baseOffset: Long, maxIndexSize: Int = -1, writable:
 
   @volatile private var _lastEntry = lastEntryFromIndexFile
 
+  override def hotEntries = 340
```

Comment:
Instead of hard code the num.hot entries based on the 4KB warm section, should we make it a public config, e.g. `warm.index.section.num.bytes` and then use that to determine the num.entries here?

Reply:
If so, I will give up and close this pull request.
I don't want to make the configuration over complicated. i have explained this in the ticket. 
https://issues.apache.org/jira/browse/KAFKA-6432

If you don't like this, I will give up

Reply:
@ying-zheng I think the question here is do we see a need for this to be configurable by users based on their deployment, as mentioned in the JIRA about memory size and workload. Can this be another config that users can tune get a better performance? Either way, its good to have your comment in the ticket as code comment or if adding as a config, in config doc. So that everyone has context.